internal/engine/wazevo/backend/isa/arm64/instr_encoding.go

```diff
@@ -849,6 +928,48 @@ func encodeAluRRImm(op aluOp, rd, rn, amount, _64bit uint32) uint32 {
 	return _64bit<<31 | opc<<29 | 0b100110<<23 | _64bit<<22 | immr<<16 | imms<<10 | rn<<5 | rd
 }
 
+func encodeVecRRR(op vecOp, rd, rn uint32, arr vecArrangement) uint32 {
+	switch op {
+	case vecOpUaddlv:
+		q, size := vecArrangementParams(arr)
+		return q<<29 | 0b101110<<24 | size<<22 | 0b110000001110<<10 | rn<<5 | rd
```

Comment:
this should be the outer most return of this function. In other words, encoding rule must be the same for all vecOp passed to `encode**` function like others.